torch/masked/maskedtensor/core.py

```diff
@@ -0,0 +1,533 @@
+        self._preprocess_data(data, mask)
+        self._validate_members()
+
+    def _set_data_mask(self, data, mask):
+        # This method is regrettably necessary for in-place operations
+
+        self.masked_data = data
+        self.masked_mask = mask
+        self._validate_members()
+
+    def __repr__(self):
+        formatter = "{0:8.4f}"
+        if self.dim() == 0:
+            scalar_data = get_data(self).item()
+            data_formatted = (
+                formatter.format(scalar_data)
+                if isinstance(scalar_data, float)
+                else str(scalar_data)
+            )
+            if not get_mask(self).item():
+                data_formatted = "--"
+            return (
+                "masked_tensor("
+                + data_formatted
+                + ", "
+                + str(get_mask(self).item())
+                + ")"
+            )
+        s = masked_tensor_str(get_data(self), get_mask(self), formatter)
+        s = "\n".join("  " + si for si in s.split("\n"))
+        return "masked_tensor(\n" + s + "\n)"
+
+    # Seems like this needs to be defined before torch_dispatch to work
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if func in [torch.Tensor.where, torch.where]:
+            assert len(args) == 3
+            assert len(kwargs) == 0
+            return MaskedWhere.apply(*args)
+        if func is torch.Tensor.contiguous:
+            return MaskedContiguous.apply(args[0])
+        if func is torch.Tensor.to_dense:
+            return MaskedToDense.apply(args[0])
+        if func is torch.Tensor.to_sparse:
+            return MaskedToSparse.apply(args[0])
+        if func is torch.Tensor.to_sparse_csr:
+            return MaskedToSparseCsr.apply(args[0])
+        if not all(issubclass(cls, t) for t in types):
+            return NotImplemented
+        logging.debug("tf redispatching to td")
+        with torch._C.DisableTorchFunction():
+            ret = func(*args, **kwargs)
+            if func in get_default_nowrap_functions():
+                return ret
+            else:
+                return torch._tensor._convert(ret, cls)
+
+    @classmethod
+    def unary(cls, fn, data, mask):
+        return MaskedTensor(fn(data), mask)
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+        if func is torch.ops.aten.stride:
+            return None
+
+        func = func.overloadpacket
+
+        assert len(args) > 0
+        if func in [torch.ops.aten.mm, torch.ops.aten.bmm]:
+            len(args) == 2
+            len(kwargs) == 0
+            return cls.matmul(args[0], args[1], func)  # type: ignore[call-arg]
+        # Doesn't work for addmm where the first argument is a Tensor
+        data = get_data(args[0])
+        mask = get_mask(args[0])
+        if func is torch.ops.aten.stride:
```

Comment:
We called `func = func.overloadpacket`, but I suspect that the first is not necessary, will test

Reply:
Ho yes, the first will never be true as you will never match the packet.
IIRC it was buggy in core for a bit where the packet was passed instead of the overload. So that might be the historical reason for the two existing. But it is not needed anymore.